### PR TITLE
Fee Discounts page redesign

### PR DESCRIPTION
**Pull request by afterburn**
This PR closes [#59](https://app.zenhub.com/workspaces/product-engineering-team-6228558038ca2f002011c6a3/issues/injectivelabs/product-zenhub-workspace/59)

[FIGMA DESIGN FILE](https://www.figma.com/file/qJtXzfX9nzh0c7yA6G9iI3/Injective-DEX?node-id=5053%3A8645)

image
image

### CODE REVIEW

**Comment by afterburn** [comment]
Ready for review, but don't merge until [this PR](https://github.com/InjectiveLabs/injective-ts/pull/42) has been merged.

File: components/partials/fee-discounts/overview.vue
Comment (bangjelkoski) [inline components/partials/fee-discounts/overview.vue:77]: Way too hacky, let's do some other approach here `[...Array(3).keys()]` maybe or `Array.from(Array(3).keys())`

File: components/partials/fee-discounts/overview.vue
Comment (bangjelkoski) [inline components/partials/fee-discounts/overview.vue:80]: okay, this is fine but in general when something like this needs to happen it's better to extract it to component that has a computed property, it's much cleaner

File: components/partials/fee-discounts/overview.vue
Comment (bangjelkoski) [inline components/partials/fee-discounts/overview.vue:284]: The blank row above. The reason why is that they are different blocks, the first block is defining variables and the second one is setting a state variable.

**Review by bangjelkoski — CHANGES_REQUESTED** [review]

File: components/partials/fee-discounts/fees/tier.vue
Comment (ThomasRalee) [inline components/partials/fee-discounts/fees/tier.vue:96]: Nitpick here, lets move 6 to the constant file for e.g. `USDT_DECIMAL_PLACE: number = 6` and use it here and remove the comment

File: components/partials/fee-discounts/overview.vue
Comment (ThomasRalee) [inline components/partials/fee-discounts/overview.vue:291]: Let's move this class to tailwind the `tailwind.config.js` file, for [example](https://v2.tailwindcss.com/docs/flex#flex-values).

File: components/partials/fee-discounts/overview.vue
Comment (ThomasRalee) [inline components/partials/fee-discounts/overview.vue:129]: Lets return `string | undefined` here and use `&mdash;` in HTML.

```
<span v-if="lastUpdateTimestamp">{{ lastUpdateTimestamp }}</span>
<span v-else>&mdash;</span>
```

File: components/partials/fee-discounts/overview.vue
Comment (ThomasRalee) [inline components/partials/fee-discounts/overview.vue:142]: Lets move store variables computed properties to the top, same with `feeDiscountSchedule`

File: components/partials/fee-discounts/overview.vue
Comment (ThomasRalee) [inline components/partials/fee-discounts/overview.vue:119]: Let move this below store variables computed property

**Comment by ThomasRalee** [comment]
Can we please upload screenshots and the Figma link for PR that involves design changes?

**Comment by afterburn** [comment]
> Can we please upload screenshots and the Figma link for in the PR description, when the implementation involves design changes?

Hey @ThomasRalee I was a bit too quick to remove this PR from draft status as I still had some cleaned up code locally that I still had to push. There is a lot of new stuff/restructuring so please review again when you have the time. I also resolved your comments.

File: components/elements/horizontal-scroll-view-indicator.vue
Comment (ThomasRalee) [inline components/elements/horizontal-scroll-view-indicator.vue:15]: Let's remove `required` here

File: components/elements/horizontal-scroll-view.vue
Comment (ThomasRalee) [inline components/elements/horizontal-scroll-view.vue:12]: Let's use `index` instead of `i`

File: components/elements/horizontal-scroll-view.vue
Comment (ThomasRalee) [inline components/elements/horizontal-scroll-view.vue:37]: Let's add a line break here

File: components/partials/fee-discounts/overview.vue
Comment (ThomasRalee) [inline components/partials/fee-discounts/overview.vue:9]: Let's move the class to the custom components, for example:
```
<StakedAmount class="flex-0-full md:col-span-3" />
```

File: components/partials/fee-discounts/staked-amount.vue
Comment (ThomasRalee) [inline components/partials/fee-discounts/staked-amount.vue:36]: Let's combine the condition checks, for example:
```
if (!feeDiscountAccountInfo || !feeDiscountAccountInfo.accountInfo) {
    return ZERO_IN_BASE
}
```

File: components/partials/fee-discounts/staked-amount.vue
Comment (ThomasRalee) [inline components/partials/fee-discounts/staked-amount.vue:49]: Let's move store variables computed property to the top

File: app/utils/constants.ts
Comment (bangjelkoski) [inline app/utils/constants.ts:127]: `USDT_DECIMALS` is a better name for this variable

**Review by bangjelkoski — CHANGES_REQUESTED** [review]

**Comment by bangjelkoski** [comment]
[PR](https://github.com/InjectiveLabs/injective-ts/pull/42) merged

**Comment by Tofel** [comment]
LGTM ✅ 

Should we merge it?

**Comment by afterburn** [comment]
I'm merging this, for future reference (if any0 I got verbal approval from @bangjelkoski [here](https://injectivelabsinc.slack.com/archives/C03FM3MHHRU/p1656943413681589?thread_ts=1656921977.429509&cid=C03FM3MHHRU).

**Pull request merged**
